### ISSUE TITLE
nufmt produces corrupted code

### ISSUE THREAD

**Issue by GrabbenD** [nushell/nufmt#66]
### Describe the bug

- Each invocation of `nufmt` corrupts the code.
- `nufmt` inserts a line break at the start of the file if no shebang is present.

### Steps To Reproduce

1) `$ touch test.nu ; chmod +x test.nu`
   ```nu
   def greet [name] {
     ["hello" $name]
   }
   
   greet "world"
   ```
2) `$ cargo install --git https://github.com/nushell/nufmt`
3) `$ ~/.cargo/bin/nufmt test.nu`

### Result

_(Note: I had to insert space for line breaks to be shown with Gitbub Markdown)_

> ```nu
>  
> def greet [name]
> {["hello" $name ]}greet "world"
>  
> ```

```console
> nu test.nu

Error: nu::parser::missing_positional

  × Missing required positional argument.
   ╭─[/home/user/me/test.nu:2:17]
 1 │ 
 2 │ def greet [name]
   ·                 ▲
   ·                 ╰── missing block
 3 │ {["hello" $name ]}greet "world"
   ╰────
  help: Usage: def {flags} <def_name> <params> <block> . Use `--help` for more information.
```

---

Executing `nufmt` again results in:

> ```nu
>  
> def greet [name]
> {["hello" $name
> ]greet
> greet
> "world"
> "world"
> ```

### Expected behavior

Formatted code with a consistent structure, even across subsequent runs

### Environment

Arch Linux

```console
> rustc --version
rustc 1.82.0 (f6e511eec 2024-10-15)

> nu --version
0.100.0
```

**Comment by fdncred** [comment]
Agreed. This is why the readme says, `This project is still very much in beta. Expect bugs, inconsistent behavior. Do not use in productive nushell scripts!` I wouldn't even call it alpha or beta. It's very premature. We're looking for people to come along side us and work on it.
